beacon-chain/rpc/prysm/v1alpha1/validator/construct_generic_block.go

```diff
@@ -0,0 +1,74 @@
+package validator
+
+import (
+	"fmt"
+
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/interfaces"
+	ethpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v4/runtime/version"
+	"google.golang.org/protobuf/proto"
+)
+
+// constructGenericBeaconBlock constructs a `GenericBeaconBlock` based on the block version and other parameters.
+func (vs *Server) constructGenericBeaconBlock(sBlk interfaces.SignedBeaconBlock, blindBlobs []*ethpb.BlindedBlobSidecar, fullBlobs []*ethpb.BlobSidecar) (*ethpb.GenericBeaconBlock, error) {
+	if sBlk == nil || sBlk.Block() == nil {
+		return nil, fmt.Errorf("block cannot be nil")
+	}
+
+	blockProto, err := sBlk.Block().Proto()
+	if err != nil {
+		return nil, err
+	}
+
+	isBlinded := sBlk.IsBlinded()
+	payloadValue := sBlk.ValueInGwei()
+
+	switch sBlk.Version() {
+	case version.Deneb:
+		return vs.constructDenebBlock(blockProto, isBlinded, payloadValue, blindBlobs, fullBlobs), nil
+	case version.Capella:
+		return vs.constructCapellaBlock(blockProto, isBlinded, payloadValue), nil
+	case version.Bellatrix:
+		return vs.constructBellatrixBlock(blockProto, isBlinded, payloadValue), nil
+	case version.Altair:
+		return vs.constructAltairBlock(blockProto), nil
+	case version.Phase0:
+		return vs.constructPhase0Block(blockProto), nil
+	default:
+		return nil, fmt.Errorf("unknown block version: %d", sBlk.Version())
+	}
+}
+
+// Helper functions for constructing blocks for each version
+func (vs *Server) constructDenebBlock(blockProto proto.Message, isBlinded bool, payloadValue uint64, blindBlobs []*ethpb.BlindedBlobSidecar, fullBlobs []*ethpb.BlobSidecar) *ethpb.GenericBeaconBlock {
+	blockAndBlobs := &ethpb.BeaconBlockAndBlobsDeneb{
+		Block: blockProto.(*ethpb.BeaconBlockDeneb),
+		Blobs: fullBlobs,
+	}
+	if isBlinded {
+		return &ethpb.GenericBeaconBlock{Block: &ethpb.GenericBeaconBlock_BlindedDeneb{BlindedDeneb: &ethpb.BlindedBeaconBlockAndBlobsDeneb{Block: blockProto.(*ethpb.BlindedBeaconBlockDeneb), Blobs: blindBlobs}}, IsBlinded: true, PayloadValue: payloadValue}
+	}
+	return &ethpb.GenericBeaconBlock{Block: &ethpb.GenericBeaconBlock_Deneb{Deneb: blockAndBlobs}, IsBlinded: false, PayloadValue: payloadValue}
```

Comment:
Note to the reviewers. In the removed code, `IsBlinded` and `PayloadValue` were not defined, but `capella` has it. I  believe it was genuinely missed so I added here